File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/LocationTargeted.java
package com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies;

import java.util.Collection;

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.shapes.SinglePoint;
import com.anathema_roguelike.environment.Location;

public class LocationTargeted<T extends Targetable> extends TargetFilter<T, OriginType> {

	@SafeVarargs
	public LocationTargeted(Class<T> targetType, TargetConstraint<T, Location> ...constraints) {
		super(targetType, constraints);
	}

	@Override
	public Collection<T> getTargets(Location target) {
		return getTargetsInShape(new SinglePoint(target), target.getEnvironment(), target);
	}
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/LocationTargeted.scala
package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.constraints.TargetConstraint
import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.shapes.SinglePoint
import com.anathema_roguelike.environment.Location
import scala.reflect.runtime.universe._

class LocationTargeted[T <: Targetable : TypeTag](val constraints: TargetConstraint[T, Location]*)
  extends TargetFilter[T, Location](constraints:_*) {

  override def getTargets(target: Option[Location]): Iterable[T] = {
    target.map(t => {
      getTargetsInShape(new SinglePoint(t), t.getEnvironment, t)
    }).getOrElse(List())
  }
}